dcmgr/lib/dcmgr/models/network.rb

```diff
@@ -154,45 +154,137 @@ def ipv4_u32_dynamic_range_array
     end
 
     def add_ipv4_dynamic_range(range_begin, range_end)
-      range_begin = IPAddress::IPv4.new("#{range_begin}/#{self[:prefix]}")
-      range_end = IPAddress::IPv4.new("#{range_end}/#{self[:prefix]}")
-      test_inclusion(*validate_range_args(range_begin, range_end)) { |range, op|
-         case op
-         when :coverbegin
-           range.range_end = range_begin
-         when :coverend
-           range.range_begin = range_end
-         when :inccur
-           range.destroy
-         end
-         range.save_changes
-      }
+      range_begin_u32, range_end_u32 = validate_range_args(range_begin, range_end)
+      dhcp_range_dataset.included_ranges(range_begin_u32, range_end_u32).destroy
```

Comment:
So it immediately destroys any old ranges and then continues to do the rest of its work.

1) What happens if there is an error later in this method?

2) What happens if a new ip address is being scheduled after the old ranges are destroyed but before the new range is added? Collector could be scheduling an IP address while Dcmgr or vdc-manage is in the middle of this method.